src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -1060,12 +1059,17 @@ private BidderResponse applyBidPriceChanges(BidderResponse bidderResponse, BidRe
         final List<BidderError> errors = new ArrayList<>(seatBid.getErrors());
 
         final String adServerCurrency = bidRequest.getCur().get(0);
-        final BigDecimal priceAdjustmentFactor = bidAdjustmentForBidder(bidRequest, bidderResponse.getBidder());
 
         for (final BidderBid bidderBid : bidderBids) {
             final Bid bid = bidderBid.getBid();
             final String bidCurrency = bidderBid.getBidCurrency();
             final BigDecimal price = bid.getPrice();
+
+            final AdjustmentsMediaType mediaType =
+                    resolveImpMediaType(bid.getImpid(), requestImps, bidderBid.getType());
+
+            final BigDecimal priceAdjustmentFactor =
```

Comment:
Price adjustment can be refactored to separate method.
Too much details for `applyBidPriceChanges()`.